include/envoy/tcp/conn_pool.h

```diff
@@ -29,16 +29,22 @@ class Cancellable {
  * Reason that a pool stream could not be obtained.
  */
 enum class PoolFailureReason {
-  // A resource overflowed and policy prevented a new stream from being created.
+  // A resource overflowed and policy prevented a new connection from being created.
   Overflow,
-  // A connection failure took place and the stream could not be bound.
-  ConnectionFailure
+  // A local connection failure took place and the connection could not be bound.
+  LocalConnectionFailure,
+  // A remote connection failure took place and the connection could not be bound.
```

Comment:
when we say bound here, we mean bound to an Envoy client connection rather than a literal bind() call right?

If so could we rephrase to make that more clear?

Reply:
Will do